File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Commands/Batches/Batch.cs
using Raven.Documentation.CodeSamples.Orders;

namespace Raven.Documentation.CodeSamples.ClientApi.Commands.Batches
{
	using System.Collections.Generic;

	using Raven.Abstractions.Commands;
	using Raven.Abstractions.Data;
	using Raven.Client.Document;
	using Raven.Json.Linq;

	public class Batch
	{
		private interface IFoo
		{
			#region batch_1
			BatchResult[] Batch(IEnumerable<ICommandData> commandDatas);
			#endregion
		}

		public Batch()
		{
			using (var store = new DocumentStore())
			{
				#region batch_2
				var results = store
					.DatabaseCommands
					.Batch(new ICommandData[]
						       {
							       new PutCommandData
								       {
									       Key = "products/999",
										   Document = RavenJObject.FromObject(new Product
											{
												Name = "My Product",
												Supplier = "suppliers/999"
											}),
											Metadata = new RavenJObject()
								       }, 
								   new PutCommandData
									   {
											Key = "suppliers/999",
											Document = RavenJObject.FromObject(new Supplier
											{
												Name = "My Supplier",
											}),
											Metadata = new RavenJObject()
									   }, 
								   new DeleteCommandData
									   {
											Key = "products/2"
									   }
						       });
				#endregion
			}
		}
	}
}

File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/commands/batches/Batch.java
package net.ravendb.clientapi.commands.batches;

import java.util.Arrays;
import java.util.List;

import net.ravendb.abstractions.commands.DeleteCommandData;
import net.ravendb.abstractions.commands.ICommandData;
import net.ravendb.abstractions.commands.PutCommandData;
import net.ravendb.abstractions.data.BatchResult;
import net.ravendb.abstractions.json.linq.RavenJObject;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;
import net.ravendb.samples.northwind.Product;
import net.ravendb.samples.northwind.Supplier;


public class Batch {
  @SuppressWarnings("unused")
  private interface IFoo {
    //region batch_1
    public BatchResult[] batch(List<ICommandData> commandDatas);
    //endregion
  }

  @SuppressWarnings("unused")
  public Batch() throws Exception {
    try (IDocumentStore store = new DocumentStore()) {
      //region batch_2
      Product product1 = new Product();
      product1.setName("My Product");
      product1.setSupplier("suppliers/999");
      PutCommandData put1 = new PutCommandData("products/999", null, RavenJObject.fromObject(product1), new RavenJObject());

      Supplier supplier = new Supplier();
      supplier.setName("My Supplier");
      PutCommandData put2 = new PutCommandData("suppliers/999", null, RavenJObject.fromObject(supplier), new RavenJObject());

      DeleteCommandData delete = new DeleteCommandData("products/2", null);

      BatchResult[] results = store.getDatabaseCommands().batch(Arrays.<ICommandData> asList(put1, put2, delete));
      //endregion
    }
  }
}
